### DISCUSSION TITLE
WebKit Support - The application does not load using cypress webKit for Safari Browser only, it throws unhandled exceptions for JavaScript libraries

### DISCUSSION THREAD

**Discussion by Ranjeet-NBLY** [cypress-io/cypress#25914] [Questions and Help]
**Cypress Automation testing issue:**
**Issue:** The application does not load using Cypress WebKit for Safari Browser only, it throws unhandled exceptions for JavaScript libraries

![image](https://user-images.githubusercontent.com/126072945/220608145-78e0331d-ddfa-42ea-8cb2-f0f2eb72b5b9.png)

**Solutions Applied:**
**Root cause Identified:** Figured out that CDN of single-spa(cdn.jsdelivr.net) UI library version [single-spa@5.9.0](mailto:single-spa@5.9.0) was throwing an unhandled exception
**Solution 1:** Removed the CDN and loaded code from the application server itself using import-maps
**Result:** Application started breaking because it was not supporting the import map for single-spa

**Solution 2:** Upgraded the library version from 'single-spa@5.9.0' to 'single-spa@5.9.4(Latest)'
**Result:** Same error

**Solution 3:** Changed the CDN provider from cdn.jsdelivr.net to unpkg 
**Result:** Same error

**Note:** Application supports >=13.5 Safari versions, Application does not load properly in less than 13.5 Safari version (It does not load a few icons/buttons/images).

There was no provision in the cypress WebKit to configure Safari browser versions so we are not sure about default Safari version supported by the WebKit. It seems this WebKit is not compatible.

**Comment by portyu9** [comment]
Hi @Ranjeet-NBLY 

There is no Cypress “default Safari version” behind the WebKit option that you can map to Safari 13.5.

Cypress's WebKit support uses the **Playwright WebKit browser**, not an installed copy of Apple Safari. The version is therefore tied to the installed `playwright-webkit` package/revision rather than to a configurable Safari release such as Safari 13.5, 14, 15, etc.

The screenshot does show the failure occurring while SystemJS is loading the `single-spa` CDN resource, but that by itself does **not** prove the problem is caused by the WebKit engine corresponding to a Safari version below 13.5.

I would separate those two questions:

1. Check which `playwright-webkit` version is installed in the Cypress project.
2. Reproduce the smallest page possible using the same WebKit build and the `single-spa` import.
3. Inspect the actual JavaScript exception from that reduced case.

If your requirement is specifically “this must work in **Safari 13.5**”, Cypress WebKit cannot validate that exact Safari release. Cypress currently describes WebKit support as experimental, and it does not provide a setting for selecting a Safari version.

Please mark this answer as accepted if it helped, thank you.
